### PR TITLE
Remove segwit from getblocktemplate

### PR DESCRIPTION
**Pull request by mderasse**

### CODE REVIEW

**Review by effectsToCause — APPROVED** [review]

File: test/functional/feature_nulldummy.py [hunk 1]
```diff
@@ -39,9 +39,6 @@ class NULLDUMMYTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
         self.setup_clean_chain = True
-        # This script tests NULLDUMMY activation, which is part of the 'segwit' deployment, so we go through
```
Comment (hd4r) [inline test/functional/feature_nulldummy.py:42]: Should we remove the entire file?

File: test/functional/p2p_segwit.py [hunk 2]
```diff
@@ -187,9 +187,9 @@ def set_test_params(self):
         self.num_nodes = 3
         # This test tests SegWit both pre and post-activation, so use the normal BIP9 activation.
         self.extra_args = [
-            ["-whitelist=127.0.0.1", "-acceptnonstdtxn=1", "-segwitheight={}".format(SEGWIT_HEIGHT)],
```
Comment (hd4r) [inline test/functional/p2p_segwit.py:190]: Should we remove the entire file?

File: test/functional/feature_segwit.py [hunk 3]
```diff
@@ -55,20 +55,17 @@ def set_test_params(self):
             [
                 "-acceptnonstdtxn=1",
                 "-rpcserialversion=0",
-                "-segwitheight=432",
```
Comment (hd4r) [inline test/functional/feature_segwit.py:58]: Should we remove the entire file?

File: src/miner.cpp [hunk 4]
```diff
@@ -204,8 +204,6 @@ bool BlockAssembler::TestPackage(uint64_t packageSize, int64_t packageSigOpsCost
 
 // Perform transaction-level checks before adding to block:
 // - transaction finality (locktime)
-// - premature witness (in case segwit transactions are added to mempool before
```
Comment (hd4r) [inline src/miner.cpp:207]: Should we remove the code with `if (!fIncludeWitness && it->GetTx().HasWitness())`?

Comment (hd4r) [reply src/miner.cpp:207]: Perhaps the variable `fIncludeWitness` can be deleted completely?

Comment (mderasse) [reply src/miner.cpp:207]: I think everything about witness can be deleted. And in plan to do it, but i already have ~6 PR and i prefer to go step by step :)

File: test/functional/feature_segwit.py (hunk 3 above)
Comment (mderasse) [reply test/functional/feature_segwit.py:58]: Done

File: test/functional/feature_nulldummy.py (hunk 1 above)
Comment (mderasse) [reply test/functional/feature_nulldummy.py:42]: Done

File: test/functional/p2p_segwit.py (hunk 2 above)
Comment (mderasse) [reply test/functional/p2p_segwit.py:190]: Done

**Review by hd4r — APPROVED** [review]

**Pull request merged**